compiler/src/dmd/backend/cgcod.d

```diff
@@ -725,12 +725,12 @@ tryagain:
 @trusted
 targ_size_t alignsection(targ_size_t base, uint alignment, int bias)
 {
-    assert(cast(int)base <= 0);
+    assert(cast(long)base <= 0);
```

Comment:
`targ_size_t` is an alias to `ulong`, therefore casting to `int` gets a part of the number.